### ISSUE TITLE
lack of automated testing

### ISSUE THREAD

**Issue by Remi-Gau** [cvnlab/GLMsingle#33]
One of my concern regarding using GLMsingle is the lask of either [unit or system test ](https://the-turing-way.netlify.app/reproducible-research/testing/testing-overview.html) which creates an uncertainty whether the toolbox is giving expected results.

Tests can usually be easily implemented with external toolbox (moxunit for matlab) or packages (pytest for python).

Automating those tests (through github actions for example) can usually also prevent regression (breaking old feature A when implementing new feature B or fixing bug C). 

If there is some interest I could help with setting some of these up.

**Comment by Remi-Gau** [comment]
To be fair, GLMsingle is not unique in its lack of testing but it does not mean we can't try to raise the bar.

**Comment by kendrickkay** [comment]
Yes, thanks for the nudge. We should make our way towards this. Let me confer with @iancharest.

**Comment by kendrickkay** [comment]
Remi, Ian and I have been working through the issues/bugs/etc. on the repo.  Currently, we have run the MATLAB and Python implementations on a small test dataset, and can confirm that the two implementations give (nearly) identical results....

In terms of the larger issue of automated unit testing, we think the code is in a good state to pursue that.  If you feel motivated to work on this, we are happy to review/discuss and would be grateful for such efforts.  Let us know if you want to discuss.

**Comment by Remi-Gau** [comment]
Yup happy to give suggestions and maybe help set some foundations if you are interested. Feel free to ping me at `remi dot gau at uclouvain dot be`
